src/commands/ptal.ts

```diff
@@ -97,32 +113,39 @@ const generateReplyFromInteraction = async (description: string, github: string,
 	let content = "";
 	let pr_state: PullRequestState = 'PENDING';
 
-	let githubURL = TryParseURL(githubOption, interaction);
-	if(githubURL)
+	//github
 	{
-		const pathSections = githubURL.pathname.split("/");
-		pathSections.shift();
-		if(githubURL.hostname != "github.com" || pathSections.length != 4 || pathSections[2] != "pull" || !Number.parseInt(pathSections[3]))
+		const githubRE = /https:\/\/github\.com\/(?<ORGANISATION>[^\/]+)\/(?<REPOSITORY>[^\/]+)\/pull\/(?<NUMBER>\d+)/;
+		const otherRE = /((?<ORGANISATION>[^\/]+)\/)?(?<REPOSITORY>[^(#|\s)]+)(#|\s)(?<NUMBER>\d+)/;
+
+		let match: RegExpMatchArray | null;
+		if((match = githubOption.match(githubRE)) == null)
 		{
-			interaction.reply({content: `Please link a pull request. Format: \`https://github.com/ORGANISATION/REPOSITORY/pull/NUMBER\``, ephemeral: true});
-			return null;
+			if((match = githubOption.match(otherRE)) == null)
+			{
+				interaction.reply({content: "The github PR entered wasn't in a supported format", ephemeral: true});
+				return null;
+			}
 		}
 
-		const [owner, repo, _, id] = pathSections;
-		const pull_number = Number.parseInt(id);
+		let groups = match.groups!;
+
 		const pr_info = {
-			owner,
-			repo,
-			pull_number
+			owner: groups["ORGANISATION"] ?? "withastro",
+			repo: groups["REPOSITORY"],
```

Comment:
No default repository, but it is optional in `otherRE`

(Is it meant to be optional in `otherRE`? I would have expected for `#123` to resolve to `withastro/astro/pulls/123` but maybe that was not the intention?)

Reply:
I don’t believe it’s optional in otherRE

Reply:
I see, `/ 123` and similar matched the regex so I thought maybe it was meant to be, but looking again that is most likely not intentional.

Reply:
In that case it could be quite handy to add a list of allowed "syntaxes" to the bot message

Reply:
> I see, `/ 123` and similar matched the regex so I thought maybe it was meant to be, but looking again that is most likely not intentional.

Yeah. I’m guessing that it matches the repo name as “/“ in that case

Reply:
My comment on the other review also addresses this issue